src/MongoDB.Bson/Serialization/Serializers/BsonClassMapSerializer.cs

```diff
@@ -29,8 +29,32 @@ namespace MongoDB.Bson.Serialization
     /// <typeparam name="TClass">The type of the class.</typeparam>
     public class BsonClassMapSerializer<TClass> : SerializerBase<TClass>, IBsonIdProvider, IBsonDocumentSerializer, IBsonPolymorphicSerializer
     {
+#if NETSTANDARD1_5
+        #region static
+        private static void CheckForISupportInitializeInterface(out MethodInfo beginInitMethodInfo, out MethodInfo endInitMethodInfo)
+        {
+            var classTypeInfo = typeof(TClass).GetTypeInfo();
+            var isupportInitializeType = classTypeInfo.GetInterface("ISupportInitialize");
+            if (isupportInitializeType != null && isupportInitializeType.FullName == "System.ComponentModel.ISupportInitialize")
```

Comment:
Someone *could* create an interface with the same namespace and interface name in a different assembly, so in theory we could check which assembly the type is created in, but I think this is sufficient.